internal/graph/check_test.go

```diff
@@ -2998,3 +3005,531 @@ func TestIteratorReadStartingFromUser(t *testing.T) {
 		})
 	}
 }
+
+func TestConsumeUsersets(t *testing.T) {
+	type dsResults struct {
+		tuples []*openfgav1.TupleKey
+		err    error
+	}
+	tests := []struct {
+		name                         string
+		model                        *openfgav1.AuthorizationModel
+		tuples                       []dsResults
+		usersetsChannelResult        []usersetsChannelStruct
+		ctxCancelled                 bool
+		expectedResolveCheckResponse *ResolveCheckResponse
+		errorExpected                error
+	}{
+		{
+			name: "userset_tuple_found",
+			model: parser.MustTransformDSLToProto(`
+				model
+					schema 1.1
+				type user
+				type group
+					relations
+						define member: [user]
+				type document
+					relations
+						define viewer: [group#member]`),
+			tuples: []dsResults{
+				{
+					tuples: []*openfgav1.TupleKey{
+						tuple.NewTupleKey("group:1", "member", "user:maria"),
+						tuple.NewTupleKey("group:2", "member", "user:maria"),
+						tuple.NewTupleKey("group:3", "member", "user:maria"),
+						tuple.NewTupleKey("group:4", "member", "user:maria"),
+					},
+					err: nil,
+				},
+			},
+			usersetsChannelResult: []usersetsChannelStruct{
+				{
+					err:            nil,
+					objectRelation: "group#member",
+					objectIDs:      []string{"0", "2", "8"},
+				},
```

Comment:
As I suggested in the other PR, please use the bare minimum setup needed to make this test pass (i.e. 1 tuples, 1 struct) - i don't want to have to make the intersection in my head to see if this test is supposed to pass or fail 😄

Reply:
Agreed, in the ideal case, there wouldn't even be a typesystem involved in this test. Only that `checkAssociatedObjects` is being called with the correct parameters, nothing else.

Reply:
We can't make that assertion because `checkAssociatedObjects` is not mock-able :/